test/calibration/experiments/test_fine_amplitude.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the fine amplitude calibration experiment."""
+
+import numpy as np
+
+from qiskit.test import QiskitTestCase
+from qiskit.pulse import DriveChannel, Drag
+import qiskit.pulse as pulse
+
+from qiskit_experiments.calibration.fine_amplitude import FineAmplitude
+from qiskit_experiments.test.mock_iq_backend import MockFineAmp
+
+
+class TestFineAmpEndToEnd(QiskitTestCase):
```

Comment:
Can we add a test for `angle_per_gate==0`? It isn't the most common case but it seems to be an edge case we keep hitting, so maybe it would be good to have a test to make sure we don't add in a `1/angle_per_gate` in the future.

Reply:
I'm thinking we should in fact forbid this. Indeed, the update rule for the pulse amplitude is `angle_per_gate / (angle_per_gate + d_theta)`. This does not make sense if the `angle_per_gate` is zero.

Reply:
That seems okay to me. I think the code is compatible with `angle_per_gate==2 * pi` and I am not sure there is any practical difference for the fit result between 0 and 2 pi. Perhaps there should be a `2 pi` test to make sure something like `angle_per_gate % (2 * pi)` is not added in the future.

Reply:
Yes, see here 221989b

Reply:
Ah, okay, I suppose that is fine for now. I meant to add a unit test that runs with `angle_per_gate==2 * np.pi` but just forbidding that value until someone needs it seems okay, too.